src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs

```diff
@@ -27,17 +27,31 @@ public class PackageSpecReferenceDependencyProvider : IDependencyProvider
 
         private readonly bool _useLegacyAssetTargetFallbackBehavior;
 
+        private readonly bool _useLegacyDependencyGraphResolution = false;
+
         public PackageSpecReferenceDependencyProvider(
             IEnumerable<ExternalProjectReference> externalProjects,
             ILogger logger) :
             this(externalProjects,
-                environmentVariableReader: EnvironmentVariableWrapper.Instance)
+                environmentVariableReader: EnvironmentVariableWrapper.Instance,
+                useLegacyDependencyGraphResolution: false)
```

Comment:
The package spec has the data about the new algorithm, you shouldn't need to pass it here.